### PR TITLE
register only non empty actions

### PR DESCRIPTION
**Pull request by chfsx**

### CODE REVIEW

File: src/UI/Implementation/Component/Button/Renderer.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ protected function renderButton(Component\Button\Button $component, RendererInte
 		if ($label !== null) {
 			$tpl->setVariable("LABEL", $component->getLabel());
 		}
-		if ($component->isActive()) {
+		if ($component->isActive() && $action !== '' && $action !== '#') {
```
Comment (Amstutz) [inline src/UI/Implementation/Component/Button/Renderer.php:59]: why  $action !== '#'. This seems very wrong.

**Comment by Amstutz** [comment]
This is resulted from the discussion spawning from: https://github.com/ILIAS-eLearning/ILIAS/pull/716#issuecomment-349627271

**Comment by Amstutz** [comment]
The question is wheter we should pass '#' as action or handle '' as a no action if such a thing is needed for firing js. The proposed version here is to handle '' as no action. What do the others think.

**Comment by Amstutz** [comment]
@klees I think it is up to you, since you proposed the buttons initially. I have no strong preference. Checkout the discussion at https://github.com/ILIAS-eLearning/ILIAS/pull/704 to get the necessary background. Maybe a third option would be to leave the action completely optional but then check for $action === null and not $action == '', which is incorrect, as @chfsx points out in the other discussion.

**Comment by Amstutz** [comment]
@klees this is a rather urgent one. @klees can you give some feedback here?

**Comment by klees** [comment]
Someone needs to explain that to me. Skype is on.

**Comment by klees** [comment]
I implemented a solution I discussed with Timon per Skype: #800

Would be very interested to hear your opinions.

**Comment by Amstutz** [comment]
I closed this one, since @klees proposed a more complete version to be discussed in the next JF. See: #800 . Nevertheless @chfsx thanks for your input to make the UI Components better.

**Pull request closed without merging**